.github/workflows/test.yml

```diff
@@ -0,0 +1,76 @@
+name: Test
+
+on:
+  - push
+  - pull_request
```

Comment:
Would it make sense to be more specific here so not too many test runs are triggered?

e.g. I've seen this:

```yaml
on:
  pull_request:
    types: [opened, synchronize, reopened]
```


here:
https://github.com/formik/tsdx/blob/master/.github/workflows/nodejs.yml

Reply:
They also use https://github.com/bahmutov/npm-install which claims to use the yarn cache better/result in faster tests (haven't tested if that's true).

Reply:
> Would it make sense to be more specific here so not too many test runs are triggered?

Done!

> They also use https://github.com/bahmutov/npm-install which claims to use the yarn cache better/result in faster tests (haven't tested if that's true).

I do like to use cache too. But travis config had explicit anti-cache setting:

```ts
# Prevent potential issues
cache:
  npm: false
  yarn: false
```

If you think it is ok to enable cache, then we can.

Reply:
We should definitely try it out, the Travis config is a bit messy as I had issues testing out different puppeteer versions 😄

Reply:
Done! Added bahmutov/npm-install

Reply:
It could make sense to add `yarn list puppeteer` somewhere so we can verify that using a different puppeteer version actually works as intended. :-)

Reply:
There is:

```yaml
      - name: debug
        run: |
          yarn list --pattern "puppeteer|puppeteer-extra"
```

Reply:
I had to change it from plain `yarn list` because that is deprecated and throws warnings.

Reply:
Sample output:

```
Run yarn list --pattern "puppeteer|puppeteer-extra"
yarn list v1.22.4
├─ @cliqz/adblocker-puppeteer@1.9.2
├─ @types/puppeteer@2.0.1
├─ puppeteer-extra-plugin-adblocker@2.11.5
├─ puppeteer-extra-plugin-anonymize-ua@2.2.11
├─ puppeteer-extra-plugin-block-resources@2.2.6
├─ puppeteer-extra-plugin-click-and-wait@2.2.6
├─ puppeteer-extra-plugin-devtools@2.2.11
├─ puppeteer-extra-plugin-flash@2.2.7
├─ puppeteer-extra-plugin-font-size@2.2.6
├─ puppeteer-extra-plugin-recaptcha@3.1.13
├─ puppeteer-extra-plugin-repl@2.2.7
├─ puppeteer-extra-plugin-stealth@2.4.13
├─ puppeteer-extra-plugin-user-data-dir@2.2.6
├─ puppeteer-extra-plugin-user-preferences@2.2.6
├─ puppeteer-extra-plugin@3.1.6
├─ puppeteer-extra@3.1.12
├─ puppeteer-firefox@0.5.1
└─ puppeteer@5.0.0
Done in 2.23s.
node_modules/puppeteer-extra/dist/index.cjs.js: ASCII text, with CRLF, LF line terminators
```